src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/ApplyChangesToWorkspaceContext.cs

```diff
@@ -146,19 +145,12 @@ private void ProcessProjectBuildFailure(IProjectRuleSnapshot snapshot)
             }
         }
 
-        private void ProcessOptions(IProjectRuleSnapshot snapshot)
+        public void SetCommandLineArgumentsToBuild(IEnumerable<string> arguments)
         {
             Assumes.NotNull(_context);
 
-            // The order of options matters and we should use warnaserror- and then warnaserror+
-            // Depending which comes first, the compiler produces different diagnostics.
-            // We just need to move warnaserror+ to the end.
-            var warnaserrorPlus = snapshot.Items.FirstOrDefault(a => a.Key.Contains("/warnaserror+"));
-            IEnumerable<string> commandLineArguments = warnaserrorPlus.Key is null ? snapshot.Items.Keys :
-                    snapshot.Items.Remove(warnaserrorPlus.Key).Keys.Append(warnaserrorPlus.Key);
-
             // We just pass all options to Roslyn
-            _context.SetOptions(commandLineArguments.ToImmutableArray());
+            _context.SetOptions(arguments.ToImmutableArray());
```

Comment:
Please do the revert in a separate PR, to make this easier to review.

Reply:
main already has the revert.

This PR is a rebase on main having the revert.